Sources/StreamChatUI/Utils/ImageCDN.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+public enum ImageCrop: String {
+    case top, bottom, left, right, center
+}
+
+public enum ImageResize: String {
+    case crop, scale, fill
+}
+
+/// Caching key provider is providing customised (filtered) key to be used for caching.
+///
+/// The URL can contain extra parameters that are changing every time. Those parameters are effectively blocking the caching.
+/// By providing customised (filtered) key, the cache will work as expected.
+public protocol ImageCDN {
+    /// Customised (filtered) key for image cache.
+    /// - Parameter imageURL: URL of the image that should be customised (filtered).
+    /// - Returns: String to be used as an image cache key.
+    func cachingKey(forImage url: URL) -> String
+    
+    /// Enhance image URL with size parameters to get thumbnail
+    /// - Parameters:
+    ///   - originalURL: URL of the image to gwt the thumbnail for.
+    ///   - preferedSize: The requested thumbnail size.
+    ///   - crop: Crop type to be used for the thumbnail.
+    ///   - resize: Resize type to be used for the thumbnail.
+    func thumbnailURL(originalURL: URL, preferedSize: CGSize, crop: ImageCrop, resize: ImageResize) -> URL
```

Comment:
This func is not actually being used?

Reply:
No, not yet. I’m trying to figure out how/when to use it. We don't want to use it every time, but only when the thumbnail is required. Which I don't know yet.

Reply:
It's being used now